workspaces/ui-v2/src/store/documentationEdit.ts

```diff
@@ -0,0 +1,187 @@
+/**
+ * There's two places to edit / modify specs - documentation and diffs page, which is
+ * why there's two slices distinct from each other (in the future if we have a single
+ * flow for editing, we could consolidate the two slices)
+ */
+
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { v4 as uuidv4 } from 'uuid';
+import {
+  AddContribution,
+  CQRSCommand,
+  IForkableSpectacle,
+} from '@useoptic/spectacle';
+
+import { RootState, AppDispatch } from './root';
+import { getValidContributions } from './selectors';
+
+const saveDocumentationChanges = createAsyncThunk<
+  {},
+  {
+    spectacle: IForkableSpectacle;
+    commitMessage: string;
+    clientId: string;
+    clientSessionId: string;
+  },
+  {
+    dispatch: AppDispatch;
+    state: RootState;
+  }
+>(
+  'SAVE_DOCUMENTATION_CHANGES',
+  async ({ spectacle, commitMessage, clientId, clientSessionId }, thunkApi) => {
+    const state = thunkApi.getState();
+    // TODO fetch from spectacle
```

Comment:
Will follow this up in a subsequent PR - a spectacle query needs to be exposed which hooks up to https://github.com/opticdev/optic/pull/856